### PR TITLE
Support parsing drop edition in oracle

### PR DESCRIPTION
**Pull request by chakkk309**
Fixes #6480.

Changes proposed in this pull request:

- Add `DROP EDITION` grammar
- Support parsing `DROP EDITION` in oracle
- Adds Tests

### CODE REVIEW

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/oracle/ddl/OracleDropEditionStatement.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import org.apache.shardingsphere.sql.parser.sql.common.statement.AbstractSQLStatement;
+import org.apache.shardingsphere.sql.parser.sql.common.statement.ddl.DDLStatement;
+import org.apache.shardingsphere.sql.parser.sql.dialect.statement.oracle.OracleStatement;
+
+/**
+ * Oracle drop edition statement.
+ */
+@ToString
+public final class OracleDropEditionStatement extends AbstractSQLStatement implements DDLStatement, OracleStatement {
+}
```
Comment (yx9o) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/oracle/ddl/OracleDropEditionStatement.java:30]: Requires a blank line at the end.

File: shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/jaxb/cases/domain/statement/ddl/DropEditionStatementTestCase.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+package org.apache.shardingsphere.test.sql.parser.parameterized.jaxb.cases.domain.statement.ddl;
+
+import org.apache.shardingsphere.test.sql.parser.parameterized.jaxb.cases.domain.statement.SQLParserTestCase;
+
+/**
+ * Drop edition statement test case.
+ */
+public final class DropEditionStatementTestCase extends SQLParserTestCase {
+}
```
Comment (yx9o) [inline shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/jaxb/cases/domain/statement/ddl/DropEditionStatementTestCase.java:26]: Requires a blank line at the end，please check several other new files.

Comment (chakkk309) [reply shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/jaxb/cases/domain/statement/ddl/DropEditionStatementTestCase.java]: Thanks! I added blank lines at the end of all four new files.

File: shardingsphere-test/shardingsphere-parser-test/src/main/resources/case/ddl/drop-edition.xml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<sql-parser-test-cases>
+    <alter-database-link sql-case-id="drop_edition" />
```
Comment (yx9o) [inline shardingsphere-test/shardingsphere-parser-test/src/main/resources/case/ddl/drop-edition.xml:20]: `alter-database-link` does not match what you defined in `SQLParserTestCases`, please check.

Comment (chakkk309) [reply shardingsphere-test/shardingsphere-parser-test/src/main/resources/case/ddl/drop-edition.xml]: Thanks! I have changed it to the correct name `drop-edition`.

**Comment by strongduanmu** [comment]
@chakkk309 Please change `Fixes #6480` to `Ref #6480`.

**Review by strongduanmu — APPROVED** [review]
LGTM.

**Pull request merged**